ext/mbstring/mbstring_singlebyte.c

```diff
@@ -0,0 +1,581 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+*/
+
+#include "mbstring_singlebyte.h"
+
+#define CK(statement) do { if ((statement) < 0) return (-1); } while (0)
+
+/* Helper for single-byte encodings which use a conversion table */
+static int mbfl_conv_singlebyte_table(int c, mbfl_convert_filter *filter, int tbl_min, const unsigned short tbl[])
```

Comment:
Marking these as `inline` may be worthwhile.

Reply:
I was thinking about that, but decided to leave it to the better discretion of the compiler.

Would you like them to be marked `inline`?

Reply:
Eh, probably doesn't make sense to do this kind of micro-optimization without having some benchmarks to check effects first :)

Reply:
Yep. Glad we think alike on this one.